src/physics/em/detail/MollerEnergyDistribution.hh

```diff
@@ -0,0 +1,59 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file MollerEnergyDistribution.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Macros.hh"
+#include "base/Types.hh"
+#include "MollerBhabha.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Helper class for \c MollerBhabhaInteractor .
+ *
+ * Sample the exiting energy for Moller scattering.
+ */
+class MollerEnergyDistribution
+{
+  public:
+    // Construct with data from MollerBhabhaInteractor
+    inline CELER_FUNCTION
+    MollerEnergyDistribution(const MollerBhabhaPointers& shared,
+                             const real_type             inc_energy);
+
+    // Sample the exiting energy
+    template<class Engine>
+    inline CELER_FUNCTION real_type operator()(Engine& rng);
+
+  private:
+    // Electron mass * c^2 [MeV]
+    real_type electron_mass_c_sq_;
+    // Electron incident energy [MeV]
+    real_type inc_energy_;
+    // Total energy of the incident particle
+    real_type total_energy_;
+    // Maximum energy fraction transferred to free electron
+    real_type max_energy_fraction_;
```

Comment:
Same comments about max energy fraction and  the incoming "shared" data as with Bhabha.